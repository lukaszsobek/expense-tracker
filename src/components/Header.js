import React from "react"
import { NavLink } from "react-router-dom"
import { connect } from "react-redux"

import { startLogout } from "../actions"

export const Header = ({ startLogout }) => (
    <div>
        Expense Tracker
        <p>
            <NavLink activeClassName="nav__link--active" to="/" exact={true}>Dashboard</NavLink> -
            <NavLink activeClassName="nav__link--active" to="/create">Add Expense</NavLink> -
            <NavLink activeClassName="nav__link--active" to="/help">Help</NavLink>
            <button onClick={startLogout}>Log out</button>
        </p>
    </div>
)

const mapDispatchToProps = dispatch => ({
    startLogout: () => dispatch(startLogout)
})

export default connect(null, mapDispatchToProps)(Header)