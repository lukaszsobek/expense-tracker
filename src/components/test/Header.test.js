import React from "react"
import { shallow } from "enzyme"

import Header from "../Header"

test("renders header",() => {
    const wrapper = shallow(<Header />)
    expect(wrapper).toMatchSnapshot()
})