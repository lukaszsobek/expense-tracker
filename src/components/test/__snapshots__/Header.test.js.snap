// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders header 1`] = `
<div>
  Expense Tracker
  <p>
    <NavLink
      activeClassName="nav__link--active"
      ariaCurrent="true"
      exact={true}
      to="/"
    >
      Dashboard
    </NavLink>
     -
    <NavLink
      activeClassName="nav__link--active"
      ariaCurrent="true"
      to="/create"
    >
      Add Expense
    </NavLink>
     -
    <NavLink
      activeClassName="nav__link--active"
      ariaCurrent="true"
      to="/help"
    >
      Help
    </NavLink>
  </p>
</div>
`;
